cluster/node-pools/master-default/userdata.yaml

```diff
@@ -711,3 +714,21 @@ write_files:
         - level: Request
           omitStages:
             - "RequestReceived"
+
+{{ if eq .Cluster.ConfigItems.enable_encryption "true" }}
+  - owner: root:root
+    path: /etc/kubernetes/config/encryption-config.yaml
+    permissions: '0400'
+    content: |
+      apiVersion: apiserver.config.k8s.io/v1
+      kind: EncryptionConfiguration
+      resources:
+        - resources:
+          - secrets
+          providers:
+          - aescbc:
```

Comment:
Can we use GCM instead of CBC?

Reply:
According to [the docs](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#providers), it won't be secure enough unless we rotate the keys often. I'd just go with KMS, even though that'll require a bit more setup.

Reply:
Agreed, the docs are quite clear about that. We should keep it in mind in case we see performance issues.